src/Blocks/BlockButton.h

```diff
@@ -145,14 +145,19 @@ class cBlockButtonHandler :
 
 	virtual bool CanBeAt(cChunkInterface & a_ChunkInterface, const Vector3i a_RelPos, const cChunk & a_Chunk) override
 	{
-		auto Meta = a_Chunk.GetMeta(a_RelPos);
-		auto SupportRelPos = AddFaceDirection(a_RelPos, BlockMetaDataToBlockFace(Meta), true);
+		NIBBLETYPE Meta = a_Chunk.GetMeta(a_RelPos);
+		Vector3i SupportRelPos = AddFaceDirection(a_RelPos, BlockMetaDataToBlockFace(Meta), true);
+
 		if (!cChunkDef::IsValidHeight(SupportRelPos.y))
 		{
 			return false;
 		}
+
 		BLOCKTYPE SupportBlockType;
-		a_Chunk.UnboundedRelGetBlockType(SupportRelPos, SupportBlockType);
+		if (!a_Chunk.UnboundedRelGetBlockType(SupportRelPos, SupportBlockType))
+		{
+			return false;
```

Comment:
We shouldn't fail the check just because the chunk hasn't loaded yet. So, this should return `true` so the block isn't destroyed. Same for all other `UnboundedRelGet` failures.

Reply:
But `bool cChunk::UnboundedRelGetBlockType` would leave out the BlockType on it's undefined value (or 0 depending on compiler behaviour). Wouldn't this cause a bug of buttons being placed on air outside the chunk? or worse, statistically?

Reply:
I'm not saying don't check the return value. I'm saying to `return true`.

Reply:
Sorry I'm not really acquainted with the behaviour. wouldn't this still allow the button (and others) to be placed if true is returned? (I'll change this too). The behaviour I imagined was placing a button on an unloaded chunk edge would cause the button to just be dropped as happens when false is returned but I guess my intuition was wrong.

Reply:
literal edge case

I'm not sure how the player will manage to _place_ a button such that the support block is in an unloaded chunk though, given that they have nothing to click on?

Reply:
Imagine a button placed against a solid block in a different chunk. The chunk with the solid block can unload, leaving the button against an unloaded chunk. That button should not drop.

Reply:
Ohhh I didn't know this is called also when chunks are brought to scope and thought it only related to player's action of actually placing a block. I should take a broader look at the code.
On it

Reply:
`CanBeAt` is called by default from `cBlockHandler::Check` which is called whenever the block or one of its neighbours is updated in any way. That could include things like water flowing which doesn't require player input.